azkaban-common/src/main/java/azkaban/project/JdbcProjectLoader.java

```diff
@@ -193,17 +193,23 @@ private Project fetchProjectByName(Connection connection, String name)
         throws ProjectManagerException {
         QueryRunner runner = new QueryRunner();
         // Fetch the project
-        Project project = null;
+        Project project;
         ProjectResultHandler handler = new ProjectResultHandler();
+        // select active project from db first, if not exist, select inactive one.
+        // At most one active project with the same name exists in db.
         try {
             List<Project> projects =
                 runner.query(connection,
-                    ProjectResultHandler.SELECT_PROJECT_BY_NAME, handler, name);
+                      ProjectResultHandler.SELECT_ACTIVE_PROJECT_BY_NAME, handler, name);
```

Comment:
SELECT_ACTIVE_PROJECT_BY_NAME is used when creating a new project.
I also doubt whether we really need to return inactive project. In the future we can probably change the logic to remove inactive projects totally and only return active one. But for now, I just keep the existing logic in case something is breaking.